Framework/Crystal/src/CentroidPeaks.cpp

```diff
@@ -355,5 +363,32 @@ void CentroidPeaks::removeEdgePeaks(
   peakWS.removePeaks(std::move(badPeaks));
 }
 
+void CentroidPeaks::sizeBanks(std::string bankName, int &nCols, int &nRows) {
+  if (bankName == "None")
+    return;
+  boost::shared_ptr<const IComponent> parent =
+      inst->getComponentByName(bankName);
+  if (!parent)
+    return;
+  if (parent->type() == "RectangularDetector") {
+    boost::shared_ptr<const RectangularDetector> RDet =
+        boost::dynamic_pointer_cast<const RectangularDetector>(parent);
+
+    nCols = RDet->xpixels();
+    nRows = RDet->ypixels();
+  } else {
+    std::vector<Geometry::IComponent_const_sptr> children;
```

Comment:
I'm not entirely sure what is happening here, but this could definitely use some comments. Looks like code to handle "rectangular like" detector banks that are not `RectangularDectectors`?

Have you considered using `ComponentInfo::componentsInSubtree`? and `ComponentInfo::detectorsInSubtree` I think these could probably be used to extract the same nRows, nColumns as you are currently doing.

Reply:
Done